src/LocalNeuralNet.cpp

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "config.h"
+#include "LocalNeuralNet.hpp"
+#include "LocalNeuralNetImp.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <numeric>
+
+#include "geopm/Exception.hpp"
+#include "geopm/Helper.hpp"
+
+namespace geopm
+{
+    std::unique_ptr<LocalNeuralNet> LocalNeuralNet::make_unique(std::vector<std::pair<std::vector<std::vector<float> >, std::vector<float> > > input) {
+        return geopm::make_unique<LocalNeuralNetImp>(input);
+    }
+
+    LocalNeuralNetImp::LocalNeuralNetImp(std::vector<std::pair<std::vector<std::vector<float> >, std::vector<float> > > input)
+    {
+        if (input.size() == 0u) {
+            throw geopm::Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        std::size_t nlayers = input.size();
+
+        m_layers.resize(nlayers);
+        for (std::size_t idx = 0; idx < nlayers; idx++) {
```

Comment:
Nit:

```suggestion
        for (std::size_t idx = 0; idx < nlayers; ++idx) {
```
Same suggestion applies in other places.